
<!DOCTYPE html>

<html>
  <head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
    <title>Getting started with geovar &#8212; geovar  documentation</title>
    <link rel="stylesheet" href="../_static/alabaster.css" type="text/css" />
    <link rel="stylesheet" href="../_static/pygments.css" type="text/css" />
    <script id="documentation_options" data-url_root="../" src="../_static/documentation_options.js"></script>
    <script src="../_static/jquery.js"></script>
    <script src="../_static/underscore.js"></script>
    <script src="../_static/doctools.js"></script>
    <script src="../_static/language_data.js"></script>
    <script crossorigin="anonymous" integrity="sha256-Ae2Vz/4ePdIu6ZyI/5ZGsYnb+m0JlOmKPjt6XZ9JJkA=" src="https://cdnjs.cloudflare.com/ajax/libs/require.js/2.3.4/require.min.js"></script>
    <link rel="index" title="Index" href="../genindex.html" />
    <link rel="search" title="Search" href="../search.html" />
    <link rel="next" title="Public API: geovar package" href="../geovar.html" />
    <link rel="prev" title="Citing geovar" href="../cite.html" />
   
  <link rel="stylesheet" href="../_static/custom.css" type="text/css" />
  
  
  <meta name="viewport" content="width=device-width, initial-scale=0.9, maximum-scale=0.9" />

  </head><body>
  

    <div class="document">
      <div class="documentwrapper">
        <div class="bodywrapper">
          

          <div class="body" role="main">
            
  
<style>
/* CSS for nbsphinx extension */

/* remove conflicting styling from Sphinx themes */
div.nbinput.container,
div.nbinput.container div.prompt,
div.nbinput.container div.input_area,
div.nbinput.container div[class*=highlight],
div.nbinput.container div[class*=highlight] pre,
div.nboutput.container,
div.nboutput.container div.prompt,
div.nboutput.container div.output_area,
div.nboutput.container div[class*=highlight],
div.nboutput.container div[class*=highlight] pre {
    background: none;
    border: none;
    padding: 0 0;
    margin: 0;
    box-shadow: none;
}

/* avoid gaps between output lines */
div.nboutput.container div[class*=highlight] pre {
    line-height: normal;
}

/* input/output containers */
div.nbinput.container,
div.nboutput.container {
    display: -webkit-flex;
    display: flex;
    align-items: flex-start;
    margin: 0;
    width: 100%;
}
@media (max-width: 540px) {
    div.nbinput.container,
    div.nboutput.container {
        flex-direction: column;
    }
}

/* input container */
div.nbinput.container {
    padding-top: 5px;
}

/* last container */
div.nblast.container {
    padding-bottom: 5px;
}

/* input prompt */
div.nbinput.container div.prompt pre {
    color: #307FC1;
}

/* output prompt */
div.nboutput.container div.prompt pre {
    color: #BF5B3D;
}

/* all prompts */
div.nbinput.container div.prompt,
div.nboutput.container div.prompt {
    width: 4.5ex;
    padding-top: 5px;
    position: relative;
    user-select: none;
}

div.nbinput.container div.prompt > div,
div.nboutput.container div.prompt > div {
    position: absolute;
    right: 0;
    margin-right: 0.3ex;
}

@media (max-width: 540px) {
    div.nbinput.container div.prompt,
    div.nboutput.container div.prompt {
        width: unset;
        text-align: left;
        padding: 0.4em;
    }
    div.nboutput.container div.prompt.empty {
        padding: 0;
    }

    div.nbinput.container div.prompt > div,
    div.nboutput.container div.prompt > div {
        position: unset;
    }
}

/* disable scrollbars on prompts */
div.nbinput.container div.prompt pre,
div.nboutput.container div.prompt pre {
    overflow: hidden;
}

/* input/output area */
div.nbinput.container div.input_area,
div.nboutput.container div.output_area {
    -webkit-flex: 1;
    flex: 1;
    overflow: auto;
}
@media (max-width: 540px) {
    div.nbinput.container div.input_area,
    div.nboutput.container div.output_area {
        width: 100%;
    }
}

/* input area */
div.nbinput.container div.input_area {
    border: 1px solid #e0e0e0;
    border-radius: 2px;
    background: #f5f5f5;
}

/* override MathJax center alignment in output cells */
div.nboutput.container div[class*=MathJax] {
    text-align: left !important;
}

/* override sphinx.ext.imgmath center alignment in output cells */
div.nboutput.container div.math p {
    text-align: left;
}

/* standard error */
div.nboutput.container div.output_area.stderr {
    background: #fdd;
}

/* ANSI colors */
.ansi-black-fg { color: #3E424D; }
.ansi-black-bg { background-color: #3E424D; }
.ansi-black-intense-fg { color: #282C36; }
.ansi-black-intense-bg { background-color: #282C36; }
.ansi-red-fg { color: #E75C58; }
.ansi-red-bg { background-color: #E75C58; }
.ansi-red-intense-fg { color: #B22B31; }
.ansi-red-intense-bg { background-color: #B22B31; }
.ansi-green-fg { color: #00A250; }
.ansi-green-bg { background-color: #00A250; }
.ansi-green-intense-fg { color: #007427; }
.ansi-green-intense-bg { background-color: #007427; }
.ansi-yellow-fg { color: #DDB62B; }
.ansi-yellow-bg { background-color: #DDB62B; }
.ansi-yellow-intense-fg { color: #B27D12; }
.ansi-yellow-intense-bg { background-color: #B27D12; }
.ansi-blue-fg { color: #208FFB; }
.ansi-blue-bg { background-color: #208FFB; }
.ansi-blue-intense-fg { color: #0065CA; }
.ansi-blue-intense-bg { background-color: #0065CA; }
.ansi-magenta-fg { color: #D160C4; }
.ansi-magenta-bg { background-color: #D160C4; }
.ansi-magenta-intense-fg { color: #A03196; }
.ansi-magenta-intense-bg { background-color: #A03196; }
.ansi-cyan-fg { color: #60C6C8; }
.ansi-cyan-bg { background-color: #60C6C8; }
.ansi-cyan-intense-fg { color: #258F8F; }
.ansi-cyan-intense-bg { background-color: #258F8F; }
.ansi-white-fg { color: #C5C1B4; }
.ansi-white-bg { background-color: #C5C1B4; }
.ansi-white-intense-fg { color: #A1A6B2; }
.ansi-white-intense-bg { background-color: #A1A6B2; }

.ansi-default-inverse-fg { color: #FFFFFF; }
.ansi-default-inverse-bg { background-color: #000000; }

.ansi-bold { font-weight: bold; }
.ansi-underline { text-decoration: underline; }


div.nbinput.container div.input_area div[class*=highlight] > pre,
div.nboutput.container div.output_area div[class*=highlight] > pre,
div.nboutput.container div.output_area div[class*=highlight].math,
div.nboutput.container div.output_area.rendered_html,
div.nboutput.container div.output_area > div.output_javascript,
div.nboutput.container div.output_area:not(.rendered_html) > img{
    padding: 5px;
}

/* fix copybtn overflow problem in chromium (needed for 'sphinx_copybutton') */
div.nbinput.container div.input_area > div[class^='highlight'],
div.nboutput.container div.output_area > div[class^='highlight']{
    overflow-y: hidden;
}

/* hide copybtn icon on prompts (needed for 'sphinx_copybutton') */
.prompt a.copybtn {
    display: none;
}

/* Some additional styling taken form the Jupyter notebook CSS */
div.rendered_html table {
  border: none;
  border-collapse: collapse;
  border-spacing: 0;
  color: black;
  font-size: 12px;
  table-layout: fixed;
}
div.rendered_html thead {
  border-bottom: 1px solid black;
  vertical-align: bottom;
}
div.rendered_html tr,
div.rendered_html th,
div.rendered_html td {
  text-align: right;
  vertical-align: middle;
  padding: 0.5em 0.5em;
  line-height: normal;
  white-space: normal;
  max-width: none;
  border: none;
}
div.rendered_html th {
  font-weight: bold;
}
div.rendered_html tbody tr:nth-child(odd) {
  background: #f5f5f5;
}
div.rendered_html tbody tr:hover {
  background: rgba(66, 165, 245, 0.2);
}
</style>
<div class="section" id="Getting-started-with-geovar">
<h1>Getting started with <code class="docutils literal notranslate"><span class="pre">geovar</span></code><a class="headerlink" href="#Getting-started-with-geovar" title="Permalink to this headline">¶</a></h1>
<p>This notebook highlights an instructive example of how to generate “GeoVar”-style plots using an example dataset of 5000 randomly chosen bi-allelic variants on Chromosome 22 from the new high-coverage sequencing of the <a class="reference external" href="http://ftp.1000genomes.ebi.ac.uk/vol1/ftp/data_collections/1000G_2504_high_coverage/working/20190425_NYGC_GATK/">1000 Genomes Project from the New York Genome Center</a></p>
<div class="section" id="Imports">
<h2>Imports<a class="headerlink" href="#Imports" title="Permalink to this headline">¶</a></h2>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[1]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre>
<span></span><span class="kn">import</span> <span class="nn">numpy</span> <span class="k">as</span> <span class="nn">np</span>
<span class="kn">import</span> <span class="nn">pandas</span> <span class="k">as</span> <span class="nn">pd</span>
<span class="kn">import</span> <span class="nn">matplotlib.pyplot</span> <span class="k">as</span> <span class="nn">plt</span>
<span class="kn">import</span> <span class="nn">pkg_resources</span>

<span class="kn">from</span> <span class="nn">geovar.utils</span> <span class="kn">import</span> <span class="n">vcf_to_freq_table</span>
<span class="kn">from</span> <span class="nn">geovar</span> <span class="kn">import</span> <span class="n">GeoVar</span><span class="p">,</span> <span class="n">GeoVarPlot</span>
</pre></div>
</div>
</div>
</div>
<div class="section" id="Data">
<h2>Data<a class="headerlink" href="#Data" title="Permalink to this headline">¶</a></h2>
<p>The <code class="docutils literal notranslate"><span class="pre">geovar</span></code> package contains example frequency tables as well as a gzipped vcf dataset to illustrate how to move from a <a class="reference external" href="https://samtools.github.io/hts-specs/VCFv4.2.pdf">VCF</a> file and a population panel file to a full “GeoVar”-plot.</p>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[2]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre>
<span></span><span class="n">data_path</span> <span class="o">=</span> <span class="n">pkg_resources</span><span class="o">.</span><span class="n">resource_filename</span><span class="p">(</span><span class="s2">&quot;geovar&quot;</span><span class="p">,</span> <span class="s2">&quot;data/&quot;</span><span class="p">)</span>

<span class="c1"># Filepath to the VCF File</span>
<span class="n">vcf_file</span> <span class="o">=</span> <span class="s2">&quot;</span><span class="si">{}</span><span class="s2">/new_1kg_nygc.chr22.biallelic_snps.filt.n5000.vcf.gz&quot;</span><span class="o">.</span><span class="n">format</span><span class="p">(</span><span class="n">data_path</span><span class="p">)</span>

<span class="c1"># Filepath to the population panel file</span>
<span class="n">population_panel</span> <span class="o">=</span> <span class="s2">&quot;</span><span class="si">{}</span><span class="s2">/integrated_call_samples_v3.20130502.1kg_superpops.indiv_label&quot;</span><span class="o">.</span><span class="n">format</span><span class="p">(</span><span class="n">data_path</span><span class="p">)</span>
</pre></div>
</div>
</div>
<p>The population panel file is a two column file with the columns <code class="docutils literal notranslate"><span class="pre">sample</span></code> and <code class="docutils literal notranslate"><span class="pre">pop</span></code> separated by whitespace. The sample column must match the <code class="docutils literal notranslate"><span class="pre">sample</span></code> IDs in the VCF file. the <code class="docutils literal notranslate"><span class="pre">pop</span></code> column contains population labels</p>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[3]:
</pre></div>
</div>
<div class="input_area highlight-bash notranslate"><div class="highlight"><pre>
<span></span>%%bash -s <span class="s2">&quot;</span><span class="nv">$population_panel</span><span class="s2">&quot;</span>

head -n5 <span class="nv">$1</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
HG00096 EUR
HG00097 EUR
HG00099 EUR
HG00100 EUR
HG00101 EUR
</pre></div></div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[4]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre>
<span></span><span class="c1"># Writing out VCF to a Frequency Table</span>
<span class="n">af_df</span> <span class="o">=</span> <span class="n">vcf_to_freq_table</span><span class="p">(</span><span class="n">vcf_file</span><span class="p">,</span> <span class="n">pop_panel</span><span class="o">=</span><span class="n">population_panel</span><span class="p">,</span> <span class="n">outfile</span><span class="o">=</span><span class="s2">&quot;</span><span class="si">{}</span><span class="s2">/test.freq.csv&quot;</span><span class="o">.</span><span class="n">format</span><span class="p">(</span><span class="n">data_path</span><span class="p">),</span> <span class="n">minor_allele</span><span class="o">=</span><span class="kc">True</span><span class="p">)</span>

<span class="c1"># Print the beginning of the allele frequency table</span>
<span class="n">af_df</span><span class="o">.</span><span class="n">head</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="nboutput docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
Int64Index([0, 1], dtype=&#39;int64&#39;)
</pre></div></div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[4]:
</pre></div>
</div>
<div class="output_area rendered_html docutils container">
<div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>CHR</th>
      <th>SNP</th>
      <th>A1</th>
      <th>A2</th>
      <th>MAC</th>
      <th>MAF</th>
      <th>AFR</th>
      <th>AMR</th>
      <th>EAS</th>
      <th>EUR</th>
      <th>SAS</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>0</th>
      <td>22</td>
      <td>10662593</td>
      <td>C</td>
      <td>T</td>
      <td>1</td>
      <td>0.000201</td>
      <td>0.000759</td>
      <td>0.000000</td>
      <td>0.000000</td>
      <td>0.000000</td>
      <td>0.000000</td>
    </tr>
    <tr>
      <th>1</th>
      <td>22</td>
      <td>10664208</td>
      <td>G</td>
      <td>A</td>
      <td>38</td>
      <td>0.008137</td>
      <td>0.028963</td>
      <td>0.000000</td>
      <td>0.000000</td>
      <td>0.000000</td>
      <td>0.000000</td>
    </tr>
    <tr>
      <th>2</th>
      <td>22</td>
      <td>10666881</td>
      <td>C</td>
      <td>A</td>
      <td>1</td>
      <td>0.000218</td>
      <td>0.000000</td>
      <td>0.000000</td>
      <td>0.000000</td>
      <td>0.000000</td>
      <td>0.001104</td>
    </tr>
    <tr>
      <th>3</th>
      <td>22</td>
      <td>10670699</td>
      <td>T</td>
      <td>A</td>
      <td>1633</td>
      <td>0.354538</td>
      <td>0.228395</td>
      <td>0.379538</td>
      <td>0.501029</td>
      <td>0.259709</td>
      <td>0.447137</td>
    </tr>
    <tr>
      <th>4</th>
      <td>22</td>
      <td>10679257</td>
      <td>A</td>
      <td>T</td>
      <td>35</td>
      <td>0.007008</td>
      <td>0.025797</td>
      <td>0.001449</td>
      <td>0.000000</td>
      <td>0.000000</td>
      <td>0.000000</td>
    </tr>
  </tbody>
</table>
</div></div>
</div>
</div>
<div class="section" id="Generating-a-GeoVar-Object">
<h2>Generating a GeoVar Object<a class="headerlink" href="#Generating-a-GeoVar-Object" title="Permalink to this headline">¶</a></h2>
<p>Once we have a frequency table, the next step is to categorize each variant by its frequency in each population. The default binning scheme is to label mutations with population allele frequencies in the interval of (0,0.05] as Rare, and those with frequency above 0.05 as common. Below we will show how to change the frequency cutoff and how to add categories.</p>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[ ]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre>
<span></span>
</pre></div>
</div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[5]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre>
<span></span><span class="c1"># Creating the GeoVar Object</span>
<span class="n">geovar_test</span> <span class="o">=</span> <span class="n">GeoVar</span><span class="p">()</span>

<span class="c1"># Adding in the frequency file (all of it)</span>
<span class="n">geovar_test</span><span class="o">.</span><span class="n">add_freq_mat</span><span class="p">(</span><span class="n">freq_mat_file</span><span class="o">=</span><span class="s2">&quot;</span><span class="si">{}</span><span class="s2">/test.freq.csv&quot;</span><span class="o">.</span><span class="n">format</span><span class="p">(</span><span class="n">data_path</span><span class="p">))</span>

<span class="c1"># Generate a geovar binning with the binning we used in our paper</span>
<span class="n">geovar_test</span><span class="o">.</span><span class="n">geovar_binning</span><span class="p">()</span>

<span class="c1"># Printing details about the GeoVar object</span>
<span class="nb">print</span><span class="p">(</span><span class="n">geovar_test</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
GeoVar
number of variants: 5000
number of pops: 5
pops: AFR,AMR,EAS,EUR,SAS
allele freq bins: (0, 0),(0, 0.05),(0.05, 1.0)
</pre></div></div>
</div>
</div>
<div class="section" id="Visualization">
<h2>Visualization<a class="headerlink" href="#Visualization" title="Permalink to this headline">¶</a></h2>
<p>Using the <code class="docutils literal notranslate"><span class="pre">GeoVar</span></code> object we created in the last section, we can generate a “GeoVar”-plot</p>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[6]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre>
<span></span><span class="n">geovar_plot</span> <span class="o">=</span> <span class="n">GeoVarPlot</span><span class="p">()</span>

<span class="c1"># Adding data directly from the geovar object itself</span>
<span class="n">geovar_plot</span><span class="o">.</span><span class="n">add_data_geovar</span><span class="p">(</span><span class="n">geovar_test</span><span class="p">)</span>

<span class="c1"># Filter to remove very rare categories (only to speed up plotting)</span>
<span class="n">geovar_plot</span><span class="o">.</span><span class="n">filter_data</span><span class="p">()</span>

<span class="c1"># Adding in a colormap</span>
<span class="n">geovar_plot</span><span class="o">.</span><span class="n">add_cmap</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[7]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre>
<span></span><span class="n">fig</span><span class="p">,</span> <span class="n">ax</span> <span class="o">=</span> <span class="n">plt</span><span class="o">.</span><span class="n">subplots</span><span class="p">(</span><span class="mi">1</span><span class="p">,</span><span class="mi">1</span><span class="p">,</span><span class="n">figsize</span><span class="o">=</span><span class="p">(</span><span class="mi">2</span><span class="p">,</span><span class="mi">6</span><span class="p">))</span>
<span class="n">geovar_plot</span><span class="o">.</span><span class="n">plot_geovar</span><span class="p">(</span><span class="n">ax</span><span class="p">);</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<img alt="../_images/notebooks_getting-started_13_0.svg" src="../_images/notebooks_getting-started_13_0.svg" /></div>
</div>
<p>If we want the percentages next to each value, we can run the following:</p>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[8]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre>
<span></span><span class="n">fig</span><span class="p">,</span> <span class="n">ax</span> <span class="o">=</span> <span class="n">plt</span><span class="o">.</span><span class="n">subplots</span><span class="p">(</span><span class="mi">1</span><span class="p">,</span> <span class="mi">2</span><span class="p">,</span> <span class="n">figsize</span><span class="o">=</span><span class="p">(</span><span class="mi">4</span><span class="p">,</span><span class="mi">6</span><span class="p">),</span> <span class="n">sharey</span><span class="o">=</span><span class="kc">True</span><span class="p">)</span>

<span class="n">geovar_plot</span><span class="o">.</span><span class="n">plot_geovar</span><span class="p">(</span><span class="n">ax</span><span class="p">[</span><span class="mi">0</span><span class="p">]);</span>

<span class="n">geovar_plot</span><span class="o">.</span><span class="n">plot_percentages</span><span class="p">(</span><span class="n">ax</span><span class="p">[</span><span class="mi">1</span><span class="p">]);</span>
<span class="n">plt</span><span class="o">.</span><span class="n">tight_layout</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<img alt="../_images/notebooks_getting-started_15_0.svg" src="../_images/notebooks_getting-started_15_0.svg" /></div>
</div>
</div>
<div class="section" id="Changing-the-Binning">
<h2>Changing the Binning<a class="headerlink" href="#Changing-the-Binning" title="Permalink to this headline">¶</a></h2>
<p>Suppose that we want to distinguish between “low-frequency” ( 1% &lt; MAF &lt; 5%) and “rare” variants (MAF &lt; 1%). We can do this by generating a new binning scheme for the <code class="docutils literal notranslate"><span class="pre">GeoVar</span></code> object and then rerunning our plotting code.</p>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[9]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre>
<span></span><span class="c1"># Setting new bins</span>
<span class="n">geovar_test</span><span class="o">.</span><span class="n">generate_bins</span><span class="p">([</span><span class="mf">0.</span><span class="p">,</span> <span class="mf">0.01</span><span class="p">,</span> <span class="mf">0.05</span><span class="p">])</span>

<span class="c1"># Generating new geovar codes</span>
<span class="n">geovar_test</span><span class="o">.</span><span class="n">geovar_binning</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[10]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre>
<span></span><span class="n">geovar_plot2</span> <span class="o">=</span> <span class="n">GeoVarPlot</span><span class="p">()</span>

<span class="c1"># Adding data directly from the geovar object itself</span>
<span class="n">geovar_plot2</span><span class="o">.</span><span class="n">add_data_geovar</span><span class="p">(</span><span class="n">geovar_test</span><span class="p">)</span>

<span class="c1"># Filter to remove very rare categories (only to speed up plotting)</span>
<span class="n">geovar_plot2</span><span class="o">.</span><span class="n">filter_data</span><span class="p">()</span>

<span class="c1"># Adding in a colormap and a new set of labels since we have an additional category</span>
<span class="n">geovar_plot2</span><span class="o">.</span><span class="n">add_cmap</span><span class="p">(</span><span class="n">str_labels</span><span class="o">=</span><span class="p">[</span><span class="s1">&#39;U&#39;</span><span class="p">,</span><span class="s1">&#39;R&#39;</span><span class="p">,</span><span class="s1">&#39;L&#39;</span><span class="p">,</span><span class="s1">&#39;C&#39;</span><span class="p">],</span> <span class="n">lbl_colors</span><span class="o">=</span><span class="p">[</span><span class="s1">&#39;black&#39;</span><span class="p">,</span> <span class="s1">&#39;black&#39;</span><span class="p">,</span><span class="s1">&#39;white&#39;</span><span class="p">,</span><span class="s1">&#39;white&#39;</span><span class="p">])</span>
</pre></div>
</div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[11]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre>
<span></span><span class="n">fig</span><span class="p">,</span> <span class="n">ax</span> <span class="o">=</span> <span class="n">plt</span><span class="o">.</span><span class="n">subplots</span><span class="p">(</span><span class="mi">1</span><span class="p">,</span> <span class="mi">2</span><span class="p">,</span> <span class="n">figsize</span><span class="o">=</span><span class="p">(</span><span class="mi">4</span><span class="p">,</span><span class="mi">6</span><span class="p">),</span> <span class="n">sharey</span><span class="o">=</span><span class="kc">True</span><span class="p">)</span>

<span class="n">geovar_plot2</span><span class="o">.</span><span class="n">plot_geovar</span><span class="p">(</span><span class="n">ax</span><span class="p">[</span><span class="mi">0</span><span class="p">]);</span>

<span class="n">geovar_plot2</span><span class="o">.</span><span class="n">plot_percentages</span><span class="p">(</span><span class="n">ax</span><span class="p">[</span><span class="mi">1</span><span class="p">]);</span>

<span class="n">plt</span><span class="o">.</span><span class="n">tight_layout</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<img alt="../_images/notebooks_getting-started_19_0.svg" src="../_images/notebooks_getting-started_19_0.svg" /></div>
</div>
<p>As you can see the “L” category does not appear till the 7th category here, but it does allow for other ways to break down the frequency categories and further exploration.</p>
</div>
<div class="section" id="Reading-in-from-GeoVar-Counts-Files">
<h2>Reading in from GeoVar Counts Files<a class="headerlink" href="#Reading-in-from-GeoVar-Counts-Files" title="Permalink to this headline">¶</a></h2>
<p>For some datasets like the ~92 million variants in the <a class="reference external" href="http://ftp.1000genomes.ebi.ac.uk/vol1/ftp/data_collections/1000G_2504_high_coverage/working/20190425_NYGC_GATK/">NYGC 1000 Genomes Project</a>, the full frequency table is too large to store in memory within a given jupyter notebook instance.</p>
<p>In this case, we suggest processing the data in batches to create a geovar counts file, containing the numeric geovar code in the first column and the total count in the second column. We have included a file that we used in our analysis containing codes used in Figure 3B in Biddanda et al 2020.</p>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[12]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre>
<span></span><span class="n">geovar_counts_file</span>  <span class="o">=</span> <span class="s2">&quot;</span><span class="si">{}</span><span class="s2">/new_1kg_nyc_hg38_filt_total.biallelic_snps.superpops_amended2.ncat3x.filt_0.geodist_cnt.txt.gz&quot;</span><span class="o">.</span><span class="n">format</span><span class="p">(</span><span class="n">data_path</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[13]:
</pre></div>
</div>
<div class="input_area highlight-bash notranslate"><div class="highlight"><pre>
<span></span>%%bash -s <span class="s2">&quot;</span><span class="nv">$geovar_counts_file</span><span class="s2">&quot;</span>

<span class="c1"># NOTE: the counts file is gzipped in this case!</span>
<span class="c1"># NOTE: you may have to change gzcat -&gt; zcat on linux!</span>
gzcat <span class="nv">$1</span> <span class="p">|</span> head -n5
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
00001 12330992
00002 6731
00010 14619729
00011 342372
00012 6324
</pre></div></div>
</div>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[14]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre>
<span></span><span class="n">geovar_plot3</span> <span class="o">=</span> <span class="n">GeoVarPlot</span><span class="p">()</span>

<span class="c1"># Adding data directly from the geovar object itself</span>
<span class="n">geovar_plot3</span><span class="o">.</span><span class="n">add_text_data</span><span class="p">(</span><span class="n">geovar_counts_file</span><span class="p">,</span> <span class="n">filt_unobserved</span><span class="o">=</span><span class="kc">False</span><span class="p">)</span>

<span class="c1"># Filter to remove very rare categories (only to speed up plotting)</span>
<span class="n">geovar_plot3</span><span class="o">.</span><span class="n">filter_data</span><span class="p">()</span>

<span class="c1"># Adding in a colormap and a new set of labels since we have an additional category</span>
<span class="n">geovar_plot3</span><span class="o">.</span><span class="n">add_cmap</span><span class="p">(</span><span class="n">str_labels</span><span class="o">=</span><span class="p">[</span><span class="s1">&#39;U&#39;</span><span class="p">,</span><span class="s1">&#39;R&#39;</span><span class="p">,</span><span class="s1">&#39;C&#39;</span><span class="p">],</span> <span class="n">lbl_colors</span><span class="o">=</span><span class="p">[</span><span class="s1">&#39;black&#39;</span><span class="p">,</span> <span class="s1">&#39;black&#39;</span><span class="p">,</span><span class="s1">&#39;white&#39;</span><span class="p">])</span>
<span class="n">geovar_plot3</span><span class="o">.</span><span class="n">add_poplabels_manual</span><span class="p">(</span><span class="n">np</span><span class="o">.</span><span class="n">array</span><span class="p">([</span><span class="s1">&#39;AFR&#39;</span><span class="p">,</span><span class="s1">&#39;EUR&#39;</span><span class="p">,</span><span class="s1">&#39;SAS&#39;</span><span class="p">,</span><span class="s1">&#39;EAS&#39;</span><span class="p">,</span><span class="s1">&#39;AMR&#39;</span><span class="p">]))</span>
</pre></div>
</div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[15]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre>
<span></span><span class="n">fig</span><span class="p">,</span> <span class="n">ax</span> <span class="o">=</span> <span class="n">plt</span><span class="o">.</span><span class="n">subplots</span><span class="p">(</span><span class="mi">1</span><span class="p">,</span> <span class="mi">2</span><span class="p">,</span> <span class="n">figsize</span><span class="o">=</span><span class="p">(</span><span class="mi">4</span><span class="p">,</span><span class="mi">6</span><span class="p">),</span> <span class="n">sharey</span><span class="o">=</span><span class="kc">True</span><span class="p">)</span>
<span class="n">geovar_plot3</span><span class="o">.</span><span class="n">plot_geovar</span><span class="p">(</span><span class="n">ax</span><span class="p">[</span><span class="mi">0</span><span class="p">]);</span>

<span class="c1"># NOTE:  you can change the default fontsize parameter to make the percentages fit</span>
<span class="n">geovar_plot3</span><span class="o">.</span><span class="n">fontsize</span> <span class="o">=</span> <span class="mi">10</span>
<span class="n">geovar_plot3</span><span class="o">.</span><span class="n">plot_percentages</span><span class="p">(</span><span class="n">ax</span><span class="p">[</span><span class="mi">1</span><span class="p">]);</span>
<span class="n">plt</span><span class="o">.</span><span class="n">tight_layout</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<img alt="../_images/notebooks_getting-started_25_0.svg" src="../_images/notebooks_getting-started_25_0.svg" /></div>
</div>
</div>
</div>


          </div>
          
        </div>
      </div>
      <div class="sphinxsidebar" role="navigation" aria-label="main navigation">
        <div class="sphinxsidebarwrapper">
<h1 class="logo"><a href="../index.html">geovar</a></h1>






<p>
<iframe src="https://ghbtns.com/github-btn.html?user=aabiddanda&repo=geovar&type=watch&count=true&size=large&v=2"
  allowtransparency="true" frameborder="0" scrolling="0" width="200px" height="35px"></iframe>
</p>





<h3>Navigation</h3>
<p class="caption"><span class="caption-text">User Guide</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../install.html">Installation</a></li>
<li class="toctree-l1"><a class="reference internal" href="../cite.html">Citing <code class="docutils literal notranslate"><span class="pre">geovar</span></code></a></li>
</ul>
<p class="caption"><span class="caption-text">Tutorials</span></p>
<ul class="current">
<li class="toctree-l1 current"><a class="current reference internal" href="#">Getting started with <code class="docutils literal notranslate"><span class="pre">geovar</span></code></a></li>
</ul>
<p class="caption"><span class="caption-text">API Documentation</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../geovar.html">Public API: <code class="docutils literal notranslate"><span class="pre">geovar</span></code> package</a></li>
</ul>

<div class="relations">
<h3>Related Topics</h3>
<ul>
  <li><a href="../index.html">Documentation overview</a><ul>
      <li>Previous: <a href="../cite.html" title="previous chapter">Citing <code class="docutils literal notranslate"><span class="pre">geovar</span></code></a></li>
      <li>Next: <a href="../geovar.html" title="next chapter">Public API: <code class="docutils literal notranslate"><span class="pre">geovar</span></code> package</a></li>
  </ul></li>
</ul>
</div>
<div id="searchbox" style="display: none" role="search">
  <h3 id="searchlabel">Quick search</h3>
    <div class="searchformwrapper">
    <form class="search" action="../search.html" method="get">
      <input type="text" name="q" aria-labelledby="searchlabel" />
      <input type="submit" value="Go" />
    </form>
    </div>
</div>
<script>$('#searchbox').show(0);</script>








        </div>
      </div>
      <div class="clearer"></div>
    </div>
    <div class="footer">
      &copy;2020, Arjun Biddanda, Daniel P. Rice, John Novembre.
      
      |
      <a href="../_sources/notebooks/getting-started.ipynb.txt"
          rel="nofollow">Page source</a>
    </div>

    
    <a href="https://github.com/aabiddanda/geovar" class="github">
        <img style="position: absolute; top: 0; right: 0; border: 0;" src="https://s3.amazonaws.com/github/ribbons/forkme_right_darkblue_121621.png" alt="Fork me on GitHub"  class="github"/>
    </a>
    

    
  </body>
</html>